### PR TITLE
[config]Support config license header comment style.

### PR DESCRIPTION
**Pull request by dongzl**
Recently, I use this project to fix myself project license header, It is a GoLand project.

I find that the header comment style config at the `languages.yaml` and `styles.yaml`, and it doesn't support config.

The `Go` file license header comment style is `//`, like:

```
// Licensed to the Apache Software Foundation (ASF) under one
// ... ...
```

But I want the license header comment style is `*`, like:
```
/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * ... ...
 */
```

So, I fix code and submit this PR.

I add the config `languages` at the `.licenserc.yaml`, and the license header comment style use this file config preferentially, if this file doesn't have this config, it use the default config at the `languages.yaml`.

### CODE REVIEW

**Comment by dongzl** [comment]
Hi skywalking-eyes community, I don't understand why the project CI failure, who can help me, Thanks.

File: pkg/config/config.go [hunk 1]
```diff
@@ -18,6 +18,7 @@
 package config
 
 import (
+	"github.com/apache/skywalking-eyes/pkg/comments"
```
Comment (Superskyyy) [inline pkg/config/config.go:21]: @dongzl Please move this to line 25 and this should fix your lint, try run `gofmt -s -w pkg/config/config.go` to fix it automatically. Just a formatting issue compaint by the CI.

**Review by Superskyyy — CHANGES_REQUESTED** [review]

File: pkg/config/config.go (hunk 1 above)
Comment (dongzl) [reply pkg/config/config.go]: > @dongzl Please move this to line 25 and this should fix your lint, try run `gofmt -s -w pkg/config/config.go` to fix it automatically. Just a formatting issue compaint by the CI.

@Superskyyy  It's OK, Thanks very much.

File: pkg/config/config.go [hunk 2]
```diff
@@ -22,15 +22,17 @@ import (
...
 
 	"gopkg.in/yaml.v3"
 )
 
 type Config struct {
-	Header header.ConfigHeader `yaml:"header"`
-	Deps   deps.ConfigDeps     `yaml:"dependency"`
+	Header    header.ConfigHeader          `yaml:"header"`
+	Deps      deps.ConfigDeps              `yaml:"dependency"`
+	Languages map[string]comments.Language `yaml:"language"`
```
Comment (kezhenxu94) [inline pkg/config/config.go:35]: This config should be under `Header`

**Comment by wu-sheng** [comment]
@kezhenxu94 Please take a look.

File: .licenserc.yaml [hunk 3]
```diff
@@ -83,3 +83,18 @@ header: # `header` section is configurations for source codes license header.
...
+    aliases:
+      - golang
+    extensions:
+      - ".go"
+    tm_scope: source.go
+    ace_mode: golang
+    codemirror_mode: go
+    codemirror_mime_type: text/x-go
+    language_id: 132
+    comment_style_id: SlashAsterisk
```
Comment (kezhenxu94) [inline .licenserc.yaml:100]: Please minimize the configs, we only need some of the fields, like `comment_style_id` and `extensions`,

**Review by kezhenxu94 — CHANGES_REQUESTED** [review]
IT's better to merge the user-configured settings with the default ones (https://github.com/apache/skywalking-eyes/blob/eb0e0b091ea41213f712f622797e37526ca1e5d6/pkg/comments/config.go#L55), so you don't need to change the `fix` logic.

File: .licenserc.yaml (hunk 3 above)
Comment (dongzl) [reply .licenserc.yaml:100]: > Please minimize the configs, we only need some of the fields, like `comment_style_id` and `extensions`,

Hi @kezhenxu94 , Thanks for reply, I will fix it.

**Comment by dongzl** [comment]
> 

Hi @kezhenxu94 , I considered about this way, but read code, 
`var commentStyles = make(map[string]CommentStyle)` is private and `comments/config.go`'s `init` method can't receive param, I don't know how to merge the config content, so I use this way now.

**Comment by kezhenxu94** [comment]
> Hi @kezhenxu94 , I considered about this way, but read code,
> `var commentStyles = make(map[string]CommentStyle)` is private and `comments/config.go`'s `init` method can't receive param, I don't know how to merge the config content, so I use this way now.

Hi @dongzl , the way I proposed needs some other changes for sure, you can

- Add a method (`OverrideLanguageCommentStyle`) in https://github.com/apache/skywalking-eyes/blob/f23c9f4ebc40c6f4e612fa65ed7188118d4eefb9/pkg/comments/config.go#L107
- Invoke the `OverrideLanguageCommentStyle` in https://github.com/apache/skywalking-eyes/blob/f23c9f4ebc40c6f4e612fa65ed7188118d4eefb9/pkg/header/config.go#L102-L103

**Comment by dongzl** [comment]
> > Hi @kezhenxu94 , I considered about this way, but read code,
> > `var commentStyles = make(map[string]CommentStyle)` is private and `comments/config.go`'s `init` method can't receive param, I don't know how to merge the config content, so I use this way now.
> 
> Hi @dongzl , the way I proposed needs some other changes for sure, you can
> 
> * Add a method (`OverrideLanguageCommentStyle`) in https://github.com/apache/skywalking-eyes/blob/f23c9f4ebc40c6f4e612fa65ed7188118d4eefb9/pkg/comments/config.go#L107
> * Invoke the `OverrideLanguageCommentStyle` in https://github.com/apache/skywalking-eyes/blob/f23c9f4ebc40c6f4e612fa65ed7188118d4eefb9/pkg/header/config.go#L102-L103

Hi @kezhenxu94 , OK I understand, I will try it.

File: pkg/header/config.go [hunk 4]
```diff
@@ -104,6 +106,8 @@ func (config *ConfigHeader) Finalize() error {
 		config.Paths = []string{"**"}
 	}
 
+	comments.OverrideLanguageCommentStyle(config.Languages)
```
Comment (kezhenxu94) [inline pkg/header/config.go:109]: 👍

**Review by kezhenxu94 — APPROVED** [review]

**Comment by kezhenxu94** [comment]
@dongzl nice done! Can you open another pull request to add the usage into the documentation as well?

**Comment by dongzl** [comment]
> @dongzl nice done! Can you open another pull request to add the usage into the documentation as well?

Hi @kezhenxu94 , Thanks for your code review, I will finish documentation.

**Pull request merged**